crates/cubecl-linalg/src/matmul/components/global/load/strategy/async_buffer_maximize_slice_length.rs

```diff
@@ -80,48 +104,91 @@ impl AsyncBufferLoadingStrategy for AsyncBufferMaximizeSliceLengthLoading {
-            let nth_slice_in_buffer = unit_count * nth_slice_local + UNIT_POS;
-
-            let nth_slice = nth_slice_in_buffer + num_slices_buffer_offset;
-
-            let window: Window<MP::EI> =
-                read_view.load_window_in_stage::<G>(nth_slice, input_ident, config);
-            let mut destination: SliceMut<Line<MP::ES>> =
-                StridedTilingLayout::nth_slice::<MP::ES, G::SmmConfig>(
-                    stage,
-                    nth_slice,
-                    comptime!(input_ident.as_ident()),
-                    config.to_smm_config(),
-                );
-
-            let start = slice_buffer_offset;
-            let limit = select(
-                slice_buffer_offset < window.size,
-                slice_buffer_offset,
-                window.size,
-            );
-            let end = start + Min::min(window.size - limit, slice_length);
-
-            let src = window.slice.slice(start, end);
-            let mut dest = destination.slice_mut(start, end);
-
-            #[allow(clippy::collapsible_else_if)]
-            if comptime!(num_slices % unit_count == 0) {
-                CM::memcpy_async(mechanism, &src.try_cast_unchecked(), &mut dest);
-            } else {
-                if nth_slice_in_buffer < num_slices {
-                    CM::memcpy_async(mechanism, &src.try_cast_unchecked(), &mut dest);
-                }
-            };
+        let num_tasks = (num_slices + unit_count - 1) / unit_count;
+
+        AsyncBufferMaximizeSliceLengthJob::<MP, CM> {
+            read_view,
+            stage,
+            mechanism,
+            num_tasks,
+            unit_count,
+            num_slices_buffer_offset,
+            input_ident,
+            slice_buffer_offset,
+            slice_length,
+            num_slices,
         }
     }
 
     fn barrier_level() -> BarrierLevel {
         BarrierLevel::cube_manual(0u32)
     }
 }
+
+#[derive(CubeType, Clone, Copy)]
+pub struct AsyncBufferMaximizeSliceLengthJob<MP: MatmulPrecision, CM: CopyMechanism<MP::ES>> {
+    read_view: TensorReader<MP::EI>,
+    stage: Stage<MP::ES, StridedTilingLayout>,
+    mechanism: CM,
+
+    #[cube(comptime)]
+    num_tasks: u32,
+    #[cube(comptime)]
+    unit_count: u32,
+    #[cube(comptime)]
+    num_slices_buffer_offset: u32,
+    #[cube(comptime)]
+    input_ident: InputIdent,
+    #[cube(comptime)]
+    slice_buffer_offset: u32,
+    #[cube(comptime)]
+    slice_length: u32,
+    #[cube(comptime)]
+    num_slices: u32,
+}
+
+#[cube]
+impl<MP: MatmulPrecision, CM: CopyMechanism<MP::ES>> LoadingJob<MP>
+    for AsyncBufferMaximizeSliceLengthJob<MP, CM>
+{
+    fn len(this: &Self) -> u32 {
+        this.num_tasks.runtime()
+    }
+
+    fn execute_task<G: GlobalConfig>(this: &mut Self, task_id: u32, #[comptime] config: G) {
+        let nth_slice_in_buffer = this.unit_count * task_id + UNIT_POS;
+
+        let nth_slice = nth_slice_in_buffer + this.num_slices_buffer_offset;
+
```

Comment:
What is the difference between `nth_slice` and `nth_slice_in_buffer`. Also what is `num_slices_buffer_offset`. 

Maybe better naming or a comment would help.